recurly/response.go

```diff
@@ -16,8 +16,12 @@ type (
 		// Errors holds an array of validation errors if any occurred.
 		Errors []Error
 
+		// Transaction holds the transaction returned with a transaction error.
+		// This will only be populated when a TransactionError is present.
+		Transaction Transaction
```

Comment:
Adding the `Transaction` to the struct is a bit too specific. Transaction errors also can occur when creating a new subscription, updating billing info, or processing a one time transaction. https://recurly.readme.io/v2.0/page/transaction-errors

If a 422 error occurs, the TransactionError should be populated in addition to the struct provided by each API method to be populated.